releases.go

```diff
@@ -116,35 +116,38 @@ func (s *ReleasesService) GetRelease(pid interface{}, tagName string, options ..
 	return r, resp, err
 }
 
-// ReleaseAssets represents release assets in CreateRelease() options
+// CreateReleaseOptions represents CreateRelease() options.
 //
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/releases/index.html#create-a-release
-type ReleaseAssets struct {
-	Links []*ReleaseAssetLink `url:"links" json:"links"`
+type CreateReleaseOptions struct {
+	Name        *string               `url:"name" json:"name"`
+	TagName     *string               `url:"tag_name" json:"tag_name"`
+	Description *string               `url:"description" json:"description"`
+	Ref         *string               `url:"ref,omitempty" json:"ref,omitempty"`
+	Milestones  []string              `url:"milestones,omitempty" json:"milestones,omitempty"`
+	Assets      *ReleaseAssetsOptions `url:"assets,omitempty" json:"assets,omitempty"`
+	ReleasedAt  *time.Time            `url:"released_at,omitempty" json:"released_at,omitempty"`
 }
 
-// ReleaseAssetLink represents release asset link in CreateRelease() options
+// ReleaseAssetsOptions represents release assets in CreateRelease() options.
 //
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/releases/index.html#create-a-release
-type ReleaseAssetLink struct {
-	Name string `url:"name" json:"name"`
-	URL  string `url:"url" json:"url"`
+type ReleaseAssetsOptions struct {
+	Links []*ReleaseAssetLinkOptions `url:"links,omitempty" json:"links,omitempty"`
```

Comment:
I still do not get why `[]*ReleaseAssetLinkOptions` and not simply `[]ReleaseAssetLinkOptions`. :-)

Reply:
For now for consistency. If we change/update that, I want to update it everywhere in the package so using any API has the same consistent behavior.